main/lsp/requests/completion.cc

```diff
@@ -1071,8 +1075,33 @@ CompletionTask::getCompletionItemForMethod(LSPTypecheckerDelegate &typechecker,
         documentation = findDocumentation(whatFile.data(gs).source(), what.data(gs)->loc().beginPos());
     }
 
-    auto prettyType = core::source_generator::prettyTypeForMethod(gs, maybeAlias, receiverType,
-                                                                  core::ShowOptions().withUseValidSyntax());
+    std::string prettyType;
+
+    if (what.data(gs)->flags.isOverloaded) {
+        vector<string> overloadTypes;
+        auto origName = what.data(gs)->name;
+        int i = 0;
+        while (true) {
+            ++i;
+            auto overloadName = gs.lookupNameUnique(core::UniqueNameKind::Overload, origName, i);
+            ENFORCE(overloadName.exists());
+            auto overload = what.data(gs)->owner.data(gs)->findMethod(gs, overloadName);
+            ENFORCE(overload.exists());
+            overloadTypes.emplace_back(
+                core::source_generator::prettyTypeForMethod(gs, overload, receiverType, core::ShowOptions()));
```

Comment:
I think we want to call `prettySigForMethod` directly, and then call `prettyDefForMethod` on the base overload

i.e., we want

```
sig { params(x: Integer).void }
sig { void }
def foo(x=...); end
```

and not what is currently in your PR description image, with the duplicated `def` and `overload.1/2` names.